src/source/ObjcGenerator.scala

```diff
@@ -178,7 +178,7 @@ class ObjcGenerator(spec: Spec) extends Generator(spec) {
     val self = marshal.typename(objcName, r)
 
     refs.header.add("#import <Foundation/Foundation.h>")
-    refs.body.add("!#import " + q(spec.objcIncludePrefix + (if (r.ext.objc) "../" else "") + marshal.headerName(ident)))
+    refs.body.add("!#import " + q(spec.objcIncludePrefix + (if (r.ext.objc) spec.objcExtendedRecordIncludePrefix else "") + marshal.headerName(ident)))
```

Comment:
Yes, but if someone sets the new argument using an absolute path (or one relative to some known root in the header search path) it'll cause a mess if they also set the objcIncludePrefix.  The old fixed string you replaced was a relative path which can safely be appended to anything.  It appears now that I've looked more carefully that the "../" prefix is really an addition to the normal prefix which would be used to include the same record if it weren't extended.  That makes sense for Dropbox's code layout where "../" was useful and no other custom prefixes are used.  But in a codebase which needs distinct values for options like objcppIncludePrefix, objcppIncludeCppPrefix, objcppIncludeObjcPrefix, etc. it's unlikely to work.

I don't think my previous suggestion would work either in that case, mind you, without a different extended record prefix which corresponds to each existing prefix option, or some sort of smarts which tries to figure out whether the two prefixes are meant to be appended or not.

Hmm, I don't actually have a great suggestion here which isn't super complicated.  I'm okay with taking the feature as-is and simply documenting the behavior as-is, if you don't want to go further.  It's useful in many cases, just not in all.